pkg/k8shandler/status.go

```diff
@@ -263,11 +262,27 @@ func (er *ElasticsearchRequest) updateNodeConditions(status *api.ElasticsearchSt
 		er.refreshDiskWatermarkThresholds()
 	}
 
+	if err := er.pruneMissingNodes(status); err != nil {
+		return err
+	}
+
+	if err := er.updatePodNodeConditions(status, thresholdEnabled); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (er *ElasticsearchRequest) pruneMissingNodes(status *api.ElasticsearchStatus) error {
+	cluster := er.cluster
 	ll := er.L()
-	for nodeIndex, nodeStatus := range status.Nodes {
-		nodeStatus := &nodeStatus
 
+	clusterNodes := nodes[nodeMapKey(er.cluster.GetName(), er.cluster.GetNamespace())]
+
+	ns := status.Nodes[:0]
+	for _, nodeStatus := range status.Nodes {
```

Comment:
the way i've found that we can update a node while iterating over its list is to do the following which i find more clean to read than your [:0] approach :
```
for index, status := range status.Nodes {
  // to change the list
  status.Nodes[index] = changedStatus
}
```

Reply:
As per offline discussion, we wan't to keep the solution as is in the PR and add the extra GC loop as discussed here: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating